src/map/script.c

```diff
@@ -12490,6 +12490,100 @@ static BUILDIN(hideonnpc)
 	return true;
 }
 
+static BUILDIN(npc_duplicate)
+{
+	int txs = -1, tys = -1;
+
+	if (script_hasdata(st, 10))
+		txs = (script_getnum(st, 10) < -1) ? -1 : script_getnum(st, 10);
+	if (script_hasdata(st, 11))
+		tys = (script_getnum(st, 11) < -1) ? -1 : script_getnum(st, 10);
+
+	if (txs == -1 && tys != -1)
+		txs = 0;
+	if (txs != -1 && tys == -1)
+		tys = 0;
+
+	const char *npc_name = script_getstr(st, 2);
+	const char *dup_name = script_getstr(st, 3);
+	const char *dup_hidden_name = script_getstr(st, 4);
+
+	char targetname[NAME_LENGTH] = "";
+	strcat(targetname, dup_name);
+	if (strcmp(dup_hidden_name, "") != 0) {
+		strncat(targetname, "#", 1);
+		strncat(targetname, dup_hidden_name, strlen(dup_hidden_name));
+	}
+
+	if (strlen(targetname) > NAME_LENGTH) {
+		ShowError("buildin_npc_duplicate: NPC name '%s' is to long (max %d chars).\n", targetname, NAME_LENGTH);
+		script_pushint(st, 0);
+		return false;
+	} else if (npc->name2id(targetname) != NULL) {
+		ShowError("buildin_npc_duplicate: NPC named '%s' already exists.\n", targetname);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	struct npc_data *nd_source = npc->name2id(npc_name);
+	int tclass_;
+	if (script_hasdata(st, 9))
+		tclass_ = (script_getnum(st, 9) < -1) ? -1 : script_getnum(st, 9);
+	else
+		tclass_ = nd_source->class_;
+
+	if (nd_source == NULL) {
+		ShowError("buildin_npc_duplicate: Source NPC not found. (%s)\n", npc_name);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	const char *tmap = script_getstr(st, 5);
+	int tmapid = map->mapname2mapid(tmap);
+	if (tmapid < 0) {
+		ShowError("buildin_npc_duplicate: Target map not found. (%s)\n", tmap);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	int tx = script_getnum(st, 6);
+	int ty = script_getnum(st, 7);
+	int tdir = script_getnum(st, 8);
+	struct npc_data *nd_target = npc->create_npc(nd_source->subtype, tmapid, tx, ty, tdir, tclass_);
+
+	safestrncpy(nd_target->name, targetname, sizeof(nd_target->name));
+	safestrncpy(nd_target->exname, targetname, sizeof(nd_target->exname));
+
+	script_pushint(st, npc->duplicate_sub(nd_target, nd_source, txs, tys, NPO_ONINIT));
```

Comment:
`npc->duplicate_sub()` returns a boolean, not an integer. Moreover the script command will return `true` even if `npc->duplicate_sub()` fails.